### ISSUE TITLE
[Suggestion] Dedicated slot(s) for maid's current tool(s)

### ISSUE THREAD

**Issue by Shionsan** [Verclene/LittleMaidReengaged#6]
I noticed that right-clicking maid with a stick makes her change her "main hand". Is it possible to add one or even two dedicated slots that would act as "maid's main tools" and decide the maid's status?
- That might allow to change maid's class without using sugar (so it would only toggle the waiting mode).
- It would also add a knight-class maid (sword&shield) possibility.
- And in tandem with [Dynamic Lights](http://atomicstryker.net/dynamiclights.php) mod and secondary slot (shield one), escorters would become even more usefull while carrying torch in their left hand, lighting up the area around them.

**Comment by Verclene** [comment]
Hmm... I hesitate to add dedicated slots. I don't want to allow to change maid modes without sugar. "Two or more trigger items" can be realized on current specification. And so on, a Torch Layer maid can light up.

**Comment by Shionsan** [comment]
> Hmm... I hesitate to add dedicated slots. I don't want to allow to change maid modes without sugar.

Changing class-logic from sugar-use to dedicated slots would demand big changes. So your hesitation is understandable. I just think having two more slots for right and left hands is a good idea. At least in terms of UI. And there is no need to give up on change-class-on-sugar-use - it's just a possibility. Like if maid would keep hitting a zombie with a bucket, because she is a fencer, but has bucket equiped in her right hand slot.

> a Torch Layer maid can light up.

The Torcher? She would use up all the torches. What I mean is not the maid who would place torches in dark areas, but for an escort-class maid that would be able to hold a torch, so it would work with Dynamic Lights mod. This mod adds light to entities (or to be correct at entity coordinates) including NPCs equipped/holding certain items (golden equipment, torches, lava bucked etc.).
In other words, if you could make an escort maid to hold an item that is in her first slot (like wild maids do) it would work nice with the Dynamic Lights mod. But with dedicated slot for the left hand it would be possible for maid to hold a torch (or any other light source item) in any mode - and that was my point.

Example: https://s23.postimg.org/mnsswaeyj/2016-04-17_20.07.53.png < a wild maid with a torch in her hand, Dynamic Lights mod makes "her" emit the light. Kinda makes torcher class useless...

Btw torch seems to be not in hand but in the air near the hand - held-item-position-bug.

**Comment by Verclene** [comment]
If you want to use maids with Dynamic Lights Mod, I can make Escorts equip an item in the first slot. And a Fencer can equip a bucket by replacing items in the first slot. So there seems to be little advantage in dedicated hand slots existing--in aspect of maid working.

**Comment by Shionsan** [comment]
Yes, that's exactly what I meant. And it would be great if you'd make that happen.

Two dedicated slots would also benefit the farmer class, as you could set specific seeds (via left hand slot) for maid to use instead of always planting wheat seeds first.

**Comment by Verclene** [comment]
Ok, I'll try to add slots in next update. But fixing known bugs will be earlier.
